app/javascript/controllers/wunderbaum_controller.js

```diff
@@ -827,5 +838,185 @@ _handleInputChange(e) {
     }
   }
 
+  _emitSelectionChange() {
+    // Get selected asset IDs (excluding folders)
+    const selectedAssetIds = this.tree.getSelectedNodes()
+      .filter(node => !node.data.folder && node.key && node.key.startsWith('a-'))
+      .map(node => parseInt(node.key.replace('a-', ''), 10))
+      .filter(id => !isNaN(id));
+
+    // Get selected folder IDs
+    const selectedFolderIds = this.tree.getSelectedNodes()
+      .filter(node => node.data.folder && node.key && !node.key.startsWith('a-'))
+      .map(node => parseInt(node.key, 10))
+      .filter(id => !isNaN(id));
+
+    // Emit custom event for batch actions controller
+    const event = new CustomEvent("wunderbaum:selectionChanged", {
+      detail: { selectedAssetIds, selectedFolderIds }
+    });
+    document.dispatchEvent(event);
+  }
 
+  // Public method to clear selection (called by batch actions after successful update)
+  clearSelection() {
+    if (this.tree) {
+      this.tree.setSelection(false);
+    }
+  }
+
+  // Public method to refresh tree display after batch updates
+  async refreshTreeDisplay(updatedAssetIds = [], updatedFolderIds = []) {
+    console.log("refreshTreeDisplay called with updatedAssetIds:", updatedAssetIds, "updatedFolderIds:", updatedFolderIds);
+    
+    // Handle updated folders first
+    if (this.tree && updatedFolderIds.length > 0) {
+      console.log("Refreshing folder nodes for updated folders:", updatedFolderIds);
```

Comment:
I think we should remove all of these console.log statements.

Reply:
Will do.